### PR TITLE
Rewrite Array.from to be spec compliant and use es abstract methods

### PR DESCRIPTION
**Pull request by JakeChampion**

### CODE REVIEW

File: polyfills/Array/from/polyfill.js [hunk 1]
```diff
@@ -1,301 +1,152 @@
...
-		}
-	};
-	var fnClass = '[object Function]';
-	var genClass = '[object GeneratorFunction]';
+			// b. If thisArg is present, let T be thisArg; else let T be undefined.
+			var thisArg = arguments.length > 2 ? arguments[2] : undefined;
+			if (thisArg !== undefined) {
+				var T = thisArg;
+			} else {
+				T = undefined;
```
Comment (notlee) [inline polyfills/Array/from/polyfill.js:34]: **var** T = undefined;
?

File: polyfills/Array/from/polyfill.js [hunk 2]
```diff
@@ -56,7 +56,7 @@
 			var k = 0;
 			// e. Repeat,
 			while (true) {
-				// i. If k ≥ 25^3-1, then
+				// i. If k ≥ 2^53-1, then
```
Comment (JakeChampion) [inline polyfills/Array/from/polyfill.js]: 👍

File: polyfills/Array/from/polyfill.js [hunk 3]
```diff
@@ -1,301 +1,152 @@
...
 					}
 
-					tempArray.push(iterableResponse.value);
-				} else if (iterableResponse.done === true) {
-					done = true;
-					break; // eslint-disable-line no-restricted-syntax
+					// vii. Else, let mappedValue be nextValue.
 				} else {
-					break; // eslint-disable-line no-restricted-syntax
+					mappedValue = nextValue;
```
[the remark this answers is not in the document]
Comment (notlee) [reply polyfills/Array/from/polyfill.js:93]: :facepalm: rookie mistake by me

File: polyfills/Array/from/polyfill.js (hunk 1 above)
Comment (notlee) [reply polyfills/Array/from/polyfill.js:34]: :facepalm: rookie mistake by me

**Review by notlee — APPROVED** [review]

**Pull request merged**